### PR TITLE
Add ingredients to search recipes

### LINKED ISSUE #21: Search recipes by ingredients, render recipes in BlobContainer
**Is your feature request related to a problem? Please describe.**
User should be able to enter available ingredients in SearchBar, then fetch recipes with those ingredients, render recipes in BlobContainer.

**Describe the solution you'd like**
1. Obtain search input, call fetch function and render recipes list.
2. Need boolean hook for conditionally render and change styling(grid columns).

### PR DESCRIPTION
**Pull request by connielion**
## Changes
1. Wrote functions for searching recipes and adding ingredients to state array.
2. Conditionally render blobs or recipe cards depending on if user clicked Find Recipes button.

## Purpose
Allow user to enter ingredients and search recipes.

## Approach
React typescript with lots of debugging.

Closes #21

### CODE REVIEW

**Review by BlackCubes — APPROVED** [review]
Looking gewd!

**Review by JorgeLVilla — APPROVED** [review]
Nice job Connie!

File: src/components/MainContainer.tsx [hunk 1]
```diff
@@ -44,13 +88,30 @@ const MainContainer = () => {
...
+      const cards = searchedRecipes.map((recipe) => {
+        const {id, title, image, missedIngredients, usedIngredients} = recipe;
+        return <RecipeCard key={id} id={id} title={title} image={image} missedIngredients={missedIngredients} usedIngredients={usedIngredients} ></RecipeCard>
+      })
+      return cards;
+    } else return [];
+  }
+ console.log(`userSearchedRecipes`, userSearchedRecipes)
+ console.log(`searchedRecipes`, searchedRecipes)
+ console.log(`cards` ,renderCards())
```
Comment (krferrell) [inline src/components/MainContainer.tsx:102]: Some console logs here. Not sure if they are needed

File: src/components/SearchBar.tsx [hunk 2]
```diff
@@ -1,51 +1,82 @@
...
+  color: #ccc;
+  border-radius: 15px;
+  border: none;
+  ::placeholder,
+  ::-webkit-input-placeholder {
+    opacity: 0.5
+  }
+  :-ms-input-placeholder {
+    font-weight: bold;
+  }
```
Comment (krferrell) [inline src/components/SearchBar.tsx:33]: good to know we can use sudo selectors in styled components

File: src/components/Sidebar.tsx [hunk 3]
```diff
@@ -27,36 +26,55 @@ const PickedIngredients = styled.div`
...
   selectedIngredients,
+  searchByIngredients,
+  setSearchedRecipes,
+  setUserSearchedRecipes
 }) => {
+  
+  const searchRecipesOnClick = async () => {
+    if (selectedIngredients.length > 0) {
+      setUserSearchedRecipes(true);
+      const recipesList = await searchByIngredients(selectedIngredients.join(','));
```
Comment (krferrell) [inline src/components/Sidebar.tsx:50]: Nice. Learned a new method. Thanks!

**Review by krferrell — APPROVED** [review]
Looking good! Keep up the good work.

**Pull request merged**